### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by vllynnyk**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+            }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    # Change error type by file
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "passed" if errors == [] else "failed"
```
Comment (YukihiroSM) [inline app/main.py:15]: you can use just `if not errors` instead of comparing to empty list.

**Review by YukihiroSM — CHANGES_REQUESTED** [review]
Also, take a look into checklist and fix your codestyle issues.

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,22 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    # Error type correction
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"
+            }
```
[the remark this answers is not in the document]
Comment (Bohdan-Salamakha) [reply app/main.py:8]: format code like this

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    # Change error type by file
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "passed" if not errors else "failed"
+            }
```
Comment (Bohdan-Salamakha) [inline app/main.py:16]:
```suggestion
    return {
            "errors": [format_linter_error(error) for error in errors],
            "path": file_path,
            "status": "passed" if not errors else "failed"
            }
```

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

**Review by Bohdan-Salamakha — APPROVED** [review]

**Review by Abnormaltype — APPROVED** [review]
Good job!

**Pull request closed without merging**